lucene/analysis/nori/src/tools/java/org/apache/lucene/analysis/ko/util/BinaryDictionaryWriter.java

```diff
@@ -137,14 +139,17 @@ public int put(String[] entry) {
       flags |= BinaryDictionary.HAS_READING;
     }
 
-    assert leftId < 8192; // there are still unused bits
-    assert posType.ordinal() < 4;
+    if (leftId >= ID_LIMIT) {
+      throw new IllegalArgumentException("leftId >= " + ID_LIMIT + ": " + leftId);
+    }
+    if (posType.ordinal() >= 4) {
+      throw new IllegalArgumentException("posType.ordinal() >= " + 4 + ": " + posType.ordinal());
+    }
     buffer.putShort((short)(leftId << 2 | posType.ordinal()));
     buffer.putShort((short) (rightId << 2 | flags));
     buffer.putShort(wordCost);
 
     if (posType == POS.Type.MORPHEME) {
-      assert leftPOS == rightPOS;
```

Comment:
I don't understand how this works in Korean - are you sure it's OK to remove?

Reply:
It is an unnecessary condition.
The leftPOS and rightPOS are determined on lines 83-90.
If posType is POS.Type.MORPHEME, leftPOS and rightPOS are always the same.

Reply:
Well, assertions are always unnecessary! I think their value is exactly in helping ensure that the correct code path was followed to get here, and confirming what we know to be true by analysis or anyway knowledge, in fact holds in practice (empirically). For example, if I were to somehow mess up the code above that guarantees this, due to my ignorance, this assertion would helpfully let me know. So I think we should keep it?

Reply:
OK :)
I understand what you mean and I'll keep it.